src/test/java/com/saralein/server/request/RequestParserTest.java

```diff
@@ -50,10 +60,21 @@ public void returnsRequestWithBody() {
             put("body", "My=Data&More=Data");
         }});
 
-        Request parsedRequest = requestParser.parse("POST /form HTTP/1.1\r\n\r\nbody: My=Data&More=Data");
+        try {
+            Request parsedRequest = requestParser.parse("POST /form HTTP/1.1\r\n\r\nbody: My=Data&More=Data");
+            assertEquals(request.getMethod(), parsedRequest.getMethod());
+            assertEquals(request.getUri(), parsedRequest.getUri());
+            assertEquals(request.getBody(), parsedRequest.getBody());
+        } catch (Exception e) {
+            fail("Failed to parse request with body.");
+        }
+    }
 
-        assertEquals(request.getMethod(), parsedRequest.getMethod());
-        assertEquals(request.getUri(), parsedRequest.getUri());
-        assertEquals(request.getBody(), parsedRequest.getBody());
+    @Test
+    public void throwsErrorForBadRequest() {
+        try {
```

Comment:
So you can use the `ExpectedException` rule here, but you can also add an argument to `@Test` which can be pretty convenient.

```java
@Test(expected = Exception.class)
public void throwsErrorForBadRequest() {
    requestParser.parse("");
}

Reply:
Nifty.  Thank you very much.  I think I saw this somewhere but didn't fully process its purpose at the time.  👍